cli/src/db.rs

```diff
@@ -0,0 +1,68 @@
+use chrono::offset::Utc;
+use dotenv::dotenv;
+use rust_decimal::prelude::{Decimal, ToPrimitive};
+use std::env;
+use tokio_postgres::{Client, Config, Error, NoTls};
+
+async fn connect() -> Result<Client, Error> {
+    dotenv().ok();
+    let pg_pass = env::var("PG_PASSWORD").expect("PG_PASSWORD must be set");
+    let pg_host = env::var("PG_HOST").expect("PG_HOST must be set");
+    let pg_db = env::var("PG_DATABASE").expect("PG_DATABASE must be set");
+    let pg_port: u16 = env::var("PG_PORT")
+        .expect("PG_PORT must be set")
+        .parse()
+        .unwrap();
+    let pg_user = env::var("PG_USER").expect("PG_USER must be set");
+
+    let (client, connection) = Config::new()
+        .password(pg_pass)
+        .host(pg_host.as_str())
+        .dbname(pg_db.as_str())
+        .port(pg_port.into())
+        .user(pg_user.as_str())
+        .application_name("Saturn Contract Payments")
+        .connect(NoTls)
+        .await
+        .unwrap();
+
+    tokio::spawn(async move {
+        if let Err(e) = connection.await {
+            eprintln!("connection error: {}", e);
+        }
+    });
+
+    Ok(client)
+}
+
+pub async fn retrieve_payments() -> Result<(Vec<String>, Vec<f64>), Error> {
+    let client = connect().await.unwrap();
+
+    let date = Utc::now();
+    let res = client
+        .query(
+            "
+       SELECT fil_wallet_address, fil_earned from payments
```

Comment:
can we make this query more generalizable ? I know some other projects are interested in using this so we may need to inject other queries. Maybe load from file or something ?

Reply:
The `retrieve_payments` function as a whole is specific to Saturn. It expects an output that formatted according to our table schema. The limiting factor here is actually the db schema. If someone has a different db schema and uses a different query, this function will fail as the query output will be wrong. The query has to be also parameterized in real time (such as date). 

I agree we should generalize our db queries but this requires a bit more effort than making the query a variable param. My suggestion is that we open an issue to track this effort separate from this PR. If you think this is highly related to the scope of this PR, we can change the implementation to make the querying much  more general.